python_modules/dagster-graphql/dagster_graphql/implementation/fetch_assets.py

```diff
@@ -71,18 +70,19 @@
     from ..schema.util import ResolveInfo
 
 
-def _normalize_asset_cursor_str(cursor_string: Optional[str]) -> Optional[str]:
+def _get_asset_key_from_cursor(cursor_string: Optional[str]) -> Optional[AssetKey]:
     # the cursor for assets is derived from a json serialized string of the path.  Because there are
     # json serialization differences between JS and Python in its treatment of whitespace, we should
     # take extra precaution here and do a deserialization/serialization pass
 
     if not cursor_string:
-        return cursor_string
+        return None
 
-    try:
-        return seven.json.dumps(seven.json.loads(cursor_string))
-    except seven.JSONDecodeError:
-        return cursor_string
+    # Filter out the initial location:repository from the id if present
+    # Assumes that '[' will never be in an asset key string after the initial character
+    cursor_string = cursor_string[cursor_string.rfind("[") :]
```

Comment:
should we be more defensive here and check for `str:str[` before mutating? just thinking about how this behaves if assumptions are invalid in the future

Reply:
the thing that I ran into here is that these are location and repository names and we don't seem to restrict "." in those names :/ (at least i can create a code location that starts with a "." locally)

Reply:
> Assumes that '[' will never be in an asset key string after the initial character

do we guard against this else where ? I was able to do 

```
>>> from dagster import AssetKey
>>> AssetKey('foo[a]')
AssetKey(['foo[a]'])
```

Reply:
Yeah I tried to use it in an asset definition and got 
```
  File "/Users/dgibson/dagster/hi.py", line 4, in <module>
    @asset(name="[hello")
     ^^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py", line 285, in inner
    return create_assets_def_from_fn_and_decorator_args(args, fn)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py", line 482, in create_assets_def_from_fn_and_decorator_args
    return builder.create_assets_definition()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/decorators/decorator_assets_definition_builder.py", line 550, in create_assets_definition
    node_def=self.create_op_definition(),
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/decorators/decorator_assets_definition_builder.py", line 531, in create_op_definition
    return _Op(
           ^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/decorators/op_decorator.py", line 125, in __call__
    op_def = OpDefinition.dagster_internal_init(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/op_definition.py", line 206, in dagster_internal_init
    return OpDefinition(
           ^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/decorator_utils.py", line 203, in wrapped_with_pre_call_fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/op_definition.py", line 183, in __init__
    super(OpDefinition, self).__init__(
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/node_definition.py", line 54, in __init__
    self._name = check_valid_name(name)
                 ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/utils.py", line 93, in check_valid_name
    check_valid_chars(name)
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/utils.py", line 101, in check_valid_chars
    raise DagsterInvalidDefinitionError(
```

I should double check that we also don't let you materialize an asset with that key without it being a definition

Reply:
maybe something like https://regex101.com/r/RDXNfK/2 if we want to do regex approach